src/nxt_http_return.c

```diff
@@ -72,13 +70,16 @@ nxt_http_action_t *
 nxt_http_return(nxt_task_t *task, nxt_http_request_t *r,
     nxt_http_action_t *action)
 {
-    nxt_http_field_t        *field;
+    nxt_int_t               ret;
+    nxt_str_t               loc;
+    nxt_str_t               *enc;
+    nxt_http_return_ctx_t   *ctx;
     nxt_http_return_conf_t  *conf;
 
     conf = action->u.conf;
 
-    nxt_debug(task, "http return: %d (loc: \"%V\")",
-              conf->status, &conf->location);
+    nxt_var_raw(conf->location, &loc);
```

Comment:
You may need to test in debug mode.

Reply:
I just tested in debug mode, and we have a problem when `"location"` contains an empty string:
```json
{
  "listeners": {
    ...,
    "*:81": {
      "pass": "routes/empty"
    },
    ...
  },
  "routes": {
    ...,
    "empty": [
      {
        "action": {
          "return": 301,
          "location": ""
        }
      }
    ],
    ...
  }
}
```